### PR TITLE
charmcraft 1.3.2 (new formula)

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by sergiusens** [comment]
I am not sure linux as a target is something doable in the short term, is there a way to skip it?

File: Formula/charmcraft.rb [hunk 1]
```diff
@@ -0,0 +1,210 @@
+class Charmcraft < Formula
+  include Language::Python::Virtualenv
+
+  desc "Shiny new formula"
+  homepage "https://charmhub.io"
+  url "https://files.pythonhosted.org/packages/41/a6/a7aac36cd53ccb5cfeefc6f2c55d5efa9c3c406a431c02905789eb102d06/charmcraft-1.3.0.tar.gz"
+  sha256 "9bfd8d366452bb832a4849c7d52e26f66f2158cf86029179540c5d594c82274b"
+  license "Apache-2.0"
+  depends_on "rust" => :build
```
Comment (carlocab) [inline Formula/charmcraft.rb:9]:
```suggestion
  license "Apache-2.0"

  depends_on "rust" => :build
```

File: Formula/charmcraft.rb [hunk 2]
```diff
@@ -0,0 +1,210 @@
+class Charmcraft < Formula
+  include Language::Python::Virtualenv
+
+  desc "Shiny new formula"
```
Comment (carlocab) [inline Formula/charmcraft.rb:4]: Can we get a slightly more informative description?

File: Formula/charmcraft.rb [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+  resource "semver" do
+    url "https://files.pythonhosted.org/packages/31/a9/b61190916030ee9af83de342e101f192bbb436c59be20a4cb0cdb7256ece/semver-2.13.0.tar.gz"
+    sha256 "fa0fe2722ee1c3f57eac478820c3a5ae2f624af8264cbdf9000c980ff7f75e3f"
+  end
+
+  resource "six" do
+    url "https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz"
+    sha256 "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926"
+  end
```
Comment (carlocab) [inline Formula/charmcraft.rb:177]: Replace this with `depends_on "six"`. You'll need to move that up with the other `depends_on` statements.

File: Formula/charmcraft.rb [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+  resource "snap-helpers" do
+    url "https://files.pythonhosted.org/packages/54/a4/7f11eb8d96826cfe17c4c68b907f8005d97ad1ef23dea67aa3d39287f6d7/snap-helpers-0.2.0.tar.gz"
+    sha256 "e72c810af32a0126bfcb7d5c39dda5ca2037889ca2fe2ee5aa543a25b9ca670b"
+  end
+
+  resource "tabulate" do
+    url "https://files.pythonhosted.org/packages/ae/3d/9d7576d94007eaf3bb685acbaaec66ff4cdeb0b18f1bf1f17edbeebffb0a/tabulate-0.8.9.tar.gz"
+    sha256 "eb1d13f25760052e8931f2ef80aaf6045a6cceb47514db8beab24cded16f13a7"
+  end
```
Comment (carlocab) [inline Formula/charmcraft.rb:187]: `depends_on "python-tabulate"`

File: Formula/charmcraft.rb [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+  def install
+    virtualenv_create(libexec, "python3")
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system "#{bin}/charmcraft", "version"
+    system "#{bin}/charmcraft", "help"
+    system "#{bin}/charmcraft", "init", "-p", testpath/"charm"
```
Comment (carlocab) [inline Formula/charmcraft.rb:207]:
```suggestion
    system bin/"charmcraft", "version"
    system bin/"charmcraft", "help"
    system bin/"charmcraft", "init", "-p", testpath/"charm"
```
Let's be consistent about not interpolating `Pathname` prefixes here.

**Review by carlocab — COMMENTED** [review]
Thanks for the PR, @sergiusens. A few comments.

File: Formula/charmcraft.rb (hunk 2 above)
Comment (sergiusens) [reply Formula/charmcraft.rb:4]: Sorry about this, not sure how I missed it

File: Formula/charmcraft.rb (hunk 4 above)
Comment (sergiusens) [reply Formula/charmcraft.rb:187]: updated

File: Formula/charmcraft.rb (hunk 5 above)
Comment (sergiusens) [reply Formula/charmcraft.rb:207]: updated

File: Formula/charmcraft.rb (hunk 3 above)
Comment (sergiusens) [reply Formula/charmcraft.rb:177]: updated

File: Formula/charmcraft.rb (hunk 1 above)
Comment (sergiusens) [reply Formula/charmcraft.rb:9]: updated

File: Formula/charmcraft.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/charmcraft.rb:4]: You seem to have missed it again

File: Formula/charmcraft.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/charmcraft.rb:9]: It's still here

File: Formula/charmcraft.rb (hunk 3 above)
Comment (carlocab) [reply Formula/charmcraft.rb:177]: Did you push the right commit?

File: Formula/charmcraft.rb (hunk 4 above)
Comment (carlocab) [reply Formula/charmcraft.rb:187]: Not yet updated, it seems.

File: Formula/charmcraft.rb (hunk 5 above)
Comment (carlocab) [reply Formula/charmcraft.rb:207]: Still needs updating.

**Comment by sergiusens** [comment]
sorry @carlocab, late night commits! I amended correctly now.

File: Formula/charmcraft.rb (hunk 1 above)
Comment (sergiusens) [reply Formula/charmcraft.rb]: done now, sorry about that

File: Formula/charmcraft.rb (hunk 2 above)
Comment (sergiusens) [reply Formula/charmcraft.rb]: done now, sorry about that

File: Formula/charmcraft.rb (hunk 3 above)
Comment (sergiusens) [reply Formula/charmcraft.rb]: I amended but forgot to `git add` :upside_down_face:

File: Formula/charmcraft.rb (hunk 5 above)
Comment (sergiusens) [reply Formula/charmcraft.rb]: also done now, again, sorry

File: Formula/charmcraft.rb [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    sha256 "50b6f157849174217d0656f99dc82fe932884fb250826c18350e159ec6cdf342"
+  end
+
+  resource "urllib3" do
+    url "https://files.pythonhosted.org/packages/4f/5a/597ef5911cb8919efe4d86206aa8b2658616d676a7088f0825ca08bd7cb8/urllib3-1.26.6.tar.gz"
+    sha256 "f57b4c16c62fa2760b7e3d97c35b255512fb6b59a259730f36ba32ce9f8e342f"
+  end
+
+  def install
+    virtualenv_create(libexec, "python3")
```
Comment (carlocab) [inline Formula/charmcraft.rb:193]:
```suggestion
```
We don't need this. We should try to fix the Linux failure too.

Comment (sergiusens) [reply Formula/charmcraft.rb]: @carlocab I removed that line, but getting linux to work is not a trivial patch that needs to be worked on upstream itself.

I also seem to be getting an audit error on "xz", seems out of scope for this PR, is that correct?

Comment (carlocab) [reply Formula/charmcraft.rb]: Yes, just rebase against `master`. This is fixed there.

**Comment by sergiusens** [comment]
@carlocab and this is now complete, thank you for your patience

**Review by SMillerDev — APPROVED** [review]
Thanks @sergiusens  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by chenrui333** [comment]
@sergiusens, I had some difficulty of upgrading the formula to use 1.5.0 release, can you help with that. 

Also [this craft_cli dependency issue](https://github.com/canonical/charmcraft/issues/617) is a bit concerning as well. Thanks!

**Comment by sergiusens** [comment]
On Wed., Mar. 9, 2022, 6:58 p.m. rui, ***@***.***> wrote:

> @sergiusens <https://github.com/sergiusens>, I had some difficulty of
> upgrading the formula to use 1.5.0 release, can you help with that.
>

I will look into it

> Also this craft_cli dependency issue
> <https://github.com/canonical/charmcraft/issues/617> is a bit concerning
> as well. Thanks!
>

It is a new dependency. There may or may not be an issue, I will sort
it out.

**Pull request closed without merging**